app/e2e/client/EnterE2EPasswordModal.tsx

```diff
@@ -0,0 +1,60 @@
+import { Box, PasswordInput, Field, FieldGroup } from '@rocket.chat/fuselage';
+import React, { FC, useState, useCallback } from 'react';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+
+import GenericModal from '../../../client/components/GenericModal';
+import { useTranslation } from '../../../client/contexts/TranslationContext';
+
+type EnterE2EPasswordModalProps = {
+	onConfirm: (password: string) => void;
+	onClose: () => void;
+};
+
+const EnterE2EPasswordModal: FC<EnterE2EPasswordModalProps> = ({ onConfirm, ...props }) => {
```

Comment:
2 Things here, that are not breaking anything, but I'd like to see:
1 - FC accepts children, but your component do not.
2 - If your modal depends on only 2 props, I don't think you should use rest notation here. Destructure only the props you need and use them. This improves legibility and also makes sure you're not forwarding unnecessary props to the component.

Reply:
For example, you're also passing the onCancel prop through the imperativeModal call, although it is not defined in the types. You could accept only the onClose prop and pass it internaly as onCancel.

Reply:
I'm passing three because the `onCancel` func is not the same as `onClose`